Assets/Scripts3D/Generator3D.cs

```diff
@@ -235,21 +244,41 @@ class Room {
         }
     }
 
-    void PlaceCube(Vector3Int location, Vector3Int size, Material material) {
+    // Floor prefab
+    void PlaceCube(Vector3Int location, Vector3Int size, Material material, string tag) {
         GameObject go = Instantiate(cubePrefab, location, Quaternion.identity);
+        // GameObject go = Instantiate(floorPrefab, location, Quaternion.identity);
         go.GetComponent<Transform>().localScale = size;
         go.GetComponent<MeshRenderer>().material = material;
+        go.tag = tag;
+    }
+
+    // Corridor prefab
+    void PlaceCorridorPrefab(Vector3Int location, Vector3Int size, Material material) {
+    GameObject go = Instantiate(cubePrefab, location, Quaternion.identity);
+    // GameObject go = Instantiate(corridorPrefab, location, Quaternion.identity);
+    go.GetComponent<Transform>().localScale = size;
+    go.GetComponent<MeshRenderer>().material = material;
+    }
+
+    // Stairs prefab
+    void StairsPrefab(Vector3Int location, Vector3Int size, Material material, string tag) {
+    GameObject go = Instantiate(cubePrefab, location, Quaternion.identity);
+    // GameObject go = Instantiate(stairsPrefab, location, Quaternion.identity);
+    go.GetComponent<Transform>().localScale = size;
+    go.GetComponent<MeshRenderer>().material = material;
     }
 
     void PlaceRoom(Vector3Int location, Vector3Int size) {
-        PlaceCube(location, size, redMaterial);
+        PlaceCube(location, size, redMaterial, roomTag);
     }
 
     void PlaceHallway(Vector3Int location) {
-        PlaceCube(location, new Vector3Int(1, 1, 1), blueMaterial);
+        PlaceCube(location, new Vector3Int(1, 1, 1), blueMaterial, hallTag);
+        // PlaceCorridorPrefab(location, new Vector3Int(1, 1, 1), blueMaterial);
```

Comment:
```suggestion
```

Reply:
Would like to keep this as commented line as its the original code for testing purposes/ reverting back